packages/runtime/src/webcontainer/terminal-config.ts

```diff
@@ -196,19 +211,24 @@ function normalizeTerminalConfig(config?: TerminalSchema): NormalizedTerminalCon
       panels.push(new TerminalPanel('terminal'));
     } else if (Array.isArray(config.panels)) {
       for (const panel of config.panels) {
-        const [type, config = {}] = Array.isArray(panel) ? panel : [panel];
-
-        let name: string | undefined;
-        let id: string | undefined;
-
-        if (typeof config === 'string') {
-          name = config;
+        let terminalPanel: TerminalPanel;
+
+        if (typeof panel === 'string') {
+          terminalPanel = new TerminalPanel(panel);
+        } else if (Array.isArray(panel)) {
+          terminalPanel = new TerminalPanel(panel[0], {
+            name: panel[1],
+          });
         } else {
-          name = config.name;
-          id = config.id;
+          terminalPanel = new TerminalPanel(panel.type, {
+            id: panel.id,
+            name: panel.name,
+            allowRedirects: panel.allowRedirects ?? config.allowRedirects,
+            allowCommands: panel.allowCommands ?? config.allowCommands,
```

Comment:
Oh this is pretty cool that we can define it once for all terminals :pinched_fingers: